include/Record.hpp

```diff
@@ -13,26 +14,45 @@ struct Record
                                                                ref_id{std::move(ref_id_i)} {}
 };
 
+/*! Used to sort two Record objects in ascending order by start position. */
 struct RecordComparatorStart
 {
-    bool operator ()(const Record & record1, const Record & record2)
+    /*!
+       \brief Compare two Record objects.
+       \param record1 The first record.
+       \param record2 The second record.
+       \return Returns `true` if record1 starts before record2, or if they both start at the same position but
+               record1 ends before record2.
```

Comment:
Shouldn't it also be true, if they start and end at the same position? Or can that case not occur?